### ISSUE TITLE
Error kali-anonsurf.deb

### ISSUE THREAD

**Issue by sniperalex117** [Und3rf10w/kali-anonsurf#21]
tried to installed with bash installer.sh
--2016-12-23 19:07:49--  https://geti2p.net/_static/i2p-debian-repo.key.asc
Resolving geti2p.net (geti2p.net)... 91.143.92.136, 2a02:180:a:65:2456:6542:1101:1010
Connecting to geti2p.net (geti2p.net)|91.143.92.136|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 14455 (14K) [text/plain]
Saving to: ‘/tmp/i2p-debian-repo.key.asc’

/tmp/i2p-debian-rep 100%[===================>]  14.12K  --.-KB/s    in 0.1s    

2016-12-23 19:07:49 (126 KB/s) - ‘/tmp/i2p-debian-repo.key.asc’ saved [14455/14455]

OK
Hit:1 http://deb.i2p2.no unstable InRelease
Hit:2 http://ftp.cc.uoc.gr/mirrors/linux/kali/kali kali-rolling InRelease   
Reading package lists... Done
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libservlet3.0-java is already the newest version (7.0.73-1).
0 upgraded, 0 newly installed, 0 to remove and 1489 not upgraded.
--2016-12-23 19:07:55--  http://ftp.us.debian.org/debian/pool/main/j/jetty8/libjetty8-java_8.1.16-4_all.deb
Resolving ftp.us.debian.org (ftp.us.debian.org)... 64.50.236.52, 128.30.2.26, 128.61.240.89, ...
Connecting to ftp.us.debian.org (ftp.us.debian.org)|64.50.236.52|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1666404 (1.6M) [application/x-troff-man]
Saving to: ‘libjetty8-java_8.1.16-4_all.deb.7’

libjetty8-java_8.1. 100%[===================>]   1.59M   513KB/s    in 3.2s    

2016-12-23 19:07:58 (513 KB/s) - ‘libjetty8-java_8.1.16-4_all.deb.7’ saved [1666404/1666404]

(Reading database ... 305027 files and directories currently installed.)
Preparing to unpack libjetty8-java_8.1.16-4_all.deb ...
Unpacking libjetty8-java (8.1.16-4) over (8.1.16-4) ...
Setting up libjetty8-java (8.1.16-4) ...
Reading package lists... Done
Building dependency tree       
Reading state information... Done
glassfish-javaee is already the newest version (1:2.1.1-b31g+dfsg1-3).
libecj-java is already the newest version (3.11.0-7).
libgetopt-java is already the newest version (1.0.14+dfsg-3).
libservlet3.0-java is already the newest version (7.0.73-1).
ttf-dejavu is already the newest version (2.37-1).
i2p is already the newest version (0.9.28-1ubuntu1).
i2p-router is already the newest version (0.9.28-1ubuntu1).
libjbigi-jni is already the newest version (0.9.28-1ubuntu1).
0 upgraded, 0 newly installed, 0 to remove and 1489 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 1489 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
i2p-keyring is already the newest version (2016.10.01).
0 upgraded, 0 newly installed, 0 to remove and 1489 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
secure-delete is already the newest version (3.1-6).
tor is already the newest version (0.2.8.11-2).
i2p is already the newest version (0.9.28-1ubuntu1).
0 upgraded, 0 newly installed, 0 to remove and 1489 not upgraded.
dpkg: error: cannot access archive 'kali-anonsurf.deb': No such file or directory
Reading package lists... Done
Building dependency tree       
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 1489 not upgraded.
dpkg: error: cannot access archive 'kali-anonsurf.deb': No such file or directory
I didn't find any answers to my problem in the previous issues, can you help me?

**Comment by Und3rf10w** [comment]
For some reason, the deb didn't build.

You can run the following commands (as root) to build the .deb and install it:
```bash
dpkg-deb -b kali-anonsurf-deb-src/ kali-anonsurf.deb # Build the deb package
dpkg -i kali-anonsurf.deb || (apt-get -f install && dpkg -i kali-anonsurf.deb) # this will automatically install the required packages
```

**Comment by sniperalex117** [comment]
these commands are in the install.sh file. 
when i run the commands it gives me this error:
control directory has bad permissions 700 (must be >=0755 and <=0775)

**Comment by Und3rf10w** [comment]
Make sure you're in the current directory for where you cloned the anonsurf package, then run:

```bash
find . -type d -exec chmod 755 {} \;
find . -type f -exec chmod 644 {} \;
```

Then rerun the previous two commands.

Did you download this via the zip, or did you do a git clone?

**Comment by sniperalex117** [comment]
i downloaded the zip

**Comment by ownmask** [comment]
Hey i am having the same issue @Und3rf10w i cant get it to install. i changed the bytes to give it the right permission @sniperalex117 (must be >=0755 and <=0775) it work but once it decide to install it comes up with errors. using kali linux. 

Selecting previously unselected package libjetty8-java.
(Reading database ... 313105 files and directories currently installed.)
Preparing to unpack libjetty8-java_8.1.16-4_all.deb ...
Unpacking libjetty8-java (8.1.16-4) ...
dpkg: dependency problems prevent configuration of libjetty8-java:
 libjetty8-java depends on libservlet3.0-java; however:
  Package libservlet3.0-java is not installed.

dpkg: error processing package libjetty8-java (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 libjetty8-java
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package libecj-java is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package libgetopt-java is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package libservlet3.0-java is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package glassfish-javaee is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package ttf-dejavu is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  fonts-dejavu fonts-dejavu-extra fonts-dejavu-core

E: Package 'libecj-java' has no installation candidate
E: Package 'libgetopt-java' has no installation candidate
E: Package 'libservlet3.0-java' has no installation candidate
E: Package 'glassfish-javaee' has no installation candidate
E: Package 'ttf-dejavu' has no installation candidate
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Correcting dependencies... Done
The following packages were automatically installed and are no longer required:
  libservice-wrapper-java libservice-wrapper-jni service-wrapper
Use 'apt autoremove' to remove them.
The following packages will be REMOVED:
  kali-anonsurf libjetty8-java
0 upgraded, 0 newly installed, 2 to remove and 0 not upgraded.
2 not fully installed or removed.
After this operation, 2,376 kB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 313348 files and directories currently installed.)
Removing kali-anonsurf (1.2.2.1) ...
Removing libjetty8-java (8.1.16-4) ...
Processing triggers for systemd (231-4) ...
Reading package lists... Done
Building dependency tree       
Reading state information... Done
i2p-keyring is already the newest version (2016.10.01).
The following packages were automatically installed and are no longer required:
  libservice-wrapper-java libservice-wrapper-jni service-wrapper
Use 'apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package tor is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package secure-delete is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'secure-delete' has no installation candidate
E: Package 'tor' has no installation candidate
dpkg-deb: building package 'kali-anonsurf' in 'kali-anonsurf.deb'.
Selecting previously unselected package kali-anonsurf.
(Reading database ... 313096 files and directories currently installed.)
Preparing to unpack kali-anonsurf.deb ...
Unpacking kali-anonsurf (1.2.2.1) ...
dpkg: dependency problems prevent configuration of kali-anonsurf:
 kali-anonsurf depends on secure-delete; however:
  Package secure-delete is not installed.
 kali-anonsurf depends on tor; however:
  Package tor is not installed.
 kali-anonsurf depends on i2p; however:
  Package i2p is not installed.

dpkg: error processing package kali-anonsurf (--install):
 dependency problems - leaving unconfigured
Processing triggers for systemd (231-4) ...
Errors were encountered while processing:
 kali-anonsurf
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Correcting dependencies... Done
The following packages were automatically installed and are no longer required:
  libservice-wrapper-java libservice-wrapper-jni service-wrapper
Use 'apt autoremove' to remove them.
The following packages will be REMOVED:
  kali-anonsurf
0 upgraded, 0 newly installed, 1 to remove and 0 not upgraded.
1 not fully installed or removed.
After this operation, 65.5 kB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 313105 files and directories currently installed.)
Removing kali-anonsurf (1.2.2.1) ...
Processing triggers for systemd (231-4) ...
Selecting previously unselected package kali-anonsurf.
(Reading database ... 313096 files and directories currently installed.)
Preparing to unpack kali-anonsurf.deb ...
Unpacking kali-anonsurf (1.2.2.1) ...
dpkg: dependency problems prevent configuration of kali-anonsurf:
 kali-anonsurf depends on secure-delete; however:
  Package secure-delete is not installed.
 kali-anonsurf depends on tor; however:
  Package tor is not installed.
 kali-anonsurf depends on i2p; however:
  Package i2p is not installed.

dpkg: error processing package kali-anonsurf (--install):
 dependency problems - leaving unconfigured
Processing triggers for systemd (231-4) ...
Errors were encountered while processing:
 kali-anonsurf

**Comment by Und3rf10w** [comment]
@sniperalex117, please delete all things anonsurf. Then attempt to ```git clone``` it.

If you do not have ```git``` installed, please do the following:

```bash
apt-get install -y git
git clone https://github.com/Und3rf10w/kali-anonsurf.git
```

When downloaded from git, it should apply the proper permissions to allow the installer to complete.

As a work around, running the following against the ```kali-anonsurf-deb``` directory should work, but I'd rather you attempted the ```git clone```:

```bash
sudo chmod 700 -R kali-anonsurf-deb
```

@ownmask, please open a separate issue. The issue you're experiencing is unrelated to this one. I'm not sure how old the version you're trying to install is, but please try the latest, and if that doesn't work, then open a new issue. You can look at issue #13 and see if that provides any helpful insight.

**Comment by ownmask** [comment]
I fix it already it. Thanks it was the permission it had to be run has root
so I changed it to 855

**Comment by sniperalex117** [comment]
Thanks for your answer.
I git cloned it but i didn't understand the "but I'd rather you attempted the git clone...". Where should I run the last command?

**Comment by Und3rf10w** [comment]
Eh, disregard that part if you git cloned it. That's just in the event that you extract it from a zip. Basically the [control directory](https://github.com/Und3rf10w/kali-anonsurf/tree/master/kali-anonsurf-deb-src/DEBIAN) doesn't have the correct permissions applied.

Does the installer work now that you've done a git clone?

**Comment by sniperalex117** [comment]
It keeps giving me the same dpkg: error: cannot access archive 'kali-anonsurf.deb': No such file or directory...

**Comment by Und3rf10w** [comment]
Please try:

```bash
dpkg-deb -b kali-anonsurf-deb-src/ kali-anonsurf.deb
```

That shouldn't output any errors.

Afterwards, run:

```bash
dpkg -i kali-anonsurf.deb || (apt-get -f install && dpkg -i kali-anonsurf.deb)
```

**Comment by mariohagi** [comment]
@sniperalex117 u can change permission on DEBIAN folder

> Owner= can change n delete

> Group= access files

> other=access files

**Comment by mariohagi** [comment]
@Und3rf10w  when i type `anonsurf start` it shows

[!] Please add the following to your /etc/default/tor and restart service:
#----------------------------------------------------------------------#
RUN_DAEMON="yes"
#----------------------------------------------------------------------#

how to fix it?
